src/charm.py

```diff
@@ -59,93 +61,96 @@ def _training_operator_layer(self) -> Layer:
         }
         return Layer(layer_config)
 
-    def _on_training_operator_pebble_ready(self, _):
-        """Placeholder for pebble ready event"""
-        pass
-
-    def _on_config_changed(self, _):
-        """Event handler for config-changed events.
-
-        On a config-changed event, a new Pebble configuration layer with the changes
-        will be created and compared to the existing one; if they differ, the Pebble
-        plan is updated and the manager service restarted.
-
-        """
-        container = self.unit.get_container(self._name)
-        # Get current config
-        current_layer = container.get_plan().services
+    def _update_layer(self) -> None:
+        """Updates the Pebble configuration layer if changed."""
+         # Get current config
+        current_layer = self._container.get_plan()
         # Create a new config layer
         new_layer = self._training_operator_layer
-        if container.can_connect():
-            # Check if there are any changes
-            if current_layer != new_layer.services:
-                container.add_layer(self._manager_service, new_layer, combine=True)
-                logging.info("Pebble plan updated with new configuration")
-                container.restart(self._manager_service)
-                logging.info("Restart training-operator")
-            self.unit.status = ActiveStatus()
-        else:
-            self.unit.status = WaitingStatus("Waiting for Pebble in workload container")
+        if current_layer.services != new_layer.services:
```

Comment:
I really haven't interacted with pebble so I don't know the answer to this, but I see the `if container.can_connect()` is removed.  I vaguely recall this being needed to protect against `.restart()`ing a container that isn't running, but if you know better ignore me :)

Reply:
In this case we're fine. You don't need to guard with `can_connect` in the `pebble-ready` event handler in general, because the event is signalling that the Juju controller was *just* able to contact the Pebble API

In this example, it looks like there is a check at the top of the `_on_config_changed` and `_on_install` that calls `can_connect` anyway :)